juju/juju
# controller-config: Allow changing (some) controller configuration post-bootstrap

## Description of change

The `controller-config` command now accepts `key=value` arguments to update controller configuration, in the same way as the `model-config` command. For the moment only audit logging configuration values can be changed - other keys might be allowed in the future (we'd need to make sure the controller handles the change correctly).

YAML config files can also be used to set the values, as with `model-config`.

This PR doesn't include changing the audit logging to respond to the updated config, that will be in a subsequent PR.

## QA steps

Bootstrap a controller with auditing-enabled=false.
See the updated usage: `juju controller-config --help`
Check that it's false: `juju controller-config auditing-enabled`
Run `juju controller-config auditing-enabled=true`
See that it's updated: `juju controller-config auditing-enabled`
Running `juju controller-config api-port=9001` fails.

## Documentation changes
Yes - documentation for `controller-config` should be updated to show changing configuration, and keys that can be changed (`auditing-enabled`, `audit-log-capture-args`, `audit-log-max-size`, `audit-log-max-backups`, `audit-log-exclude-methods`).
